Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -2418,20 +2436,7 @@ void MediaPlayerPrivateGStreamer::asyncStateChangeDone()
         if (m_isSeekPending)
             updateStates();
         else {
-            GST_DEBUG_OBJECT(pipeline(), "[Seek] seeked to %s", toString(m_seekTime).utf8().data());
-            m_isSeeking = false;
-            m_cachedPosition = MediaTime::invalidTime();
-            if (m_timeOfOverlappingSeek != m_seekTime && m_timeOfOverlappingSeek.isValid()) {
-                seek(m_timeOfOverlappingSeek);
-                m_timeOfOverlappingSeek = MediaTime::invalidTime();
-                return;
-            }
-            m_timeOfOverlappingSeek = MediaTime::invalidTime();
-
-            // The pipeline can still have a pending state. In this case a position query will fail.
-            // Right now we can use m_seekTime as a fallback.
-            m_canFallBackToLastFinishedSeekPosition = true;
-            timeChanged();
+            finishSeek();
```

Comment:
I think that this change can be refactored (I'm doing it here by myself before submitting it upstream).

When porting the change upstream and having a look at how it behaves under normal circumstances (async changes), I noticed that this finishSeek() call from asyncStateDone() was never actually called. Instead, the finishSeek() call from updateStates() was always called.

Debugging a bit more, I realized that when the async state change is completed, playbin emits both GST_MESSAGE_ASYNC_DONE and GST_MESSAGE_STATE_CHANGED. The state changed one is delivered first, processed by handleMessage(), and triggering an updateStates() call which is the one calling finishSeek() first.

With this in mind, I think that the whole asyncStateChangeDone() method is now mostly useless in the regular player. It can be emptied of content and only left there for the MSE subclass to override.